reinstall-cli/src/main.rs

```diff
@@ -0,0 +1,84 @@
+//! The main entrypoint for bootc-reinstall
+
+use std::{io::Write, process::Command};
+
+use anyhow::{Context, Result};
+
+fn run() -> Result<()> {
+    bootc_lib::cli::tracing_util::initialize_tracing();
+
+    tracing::trace!("starting bootc-reinstall");
+
+    prompt()?;
+
+    let command_and_args = [
+        // Rootless is not supported
+        "sudo",
+        // We use podman to run the bootc container. This might change in the future to remove the
+        // podman dependency.
+        "podman",
+        "run",
+        // The container needs to be privileged, as it heavily modifies the host
+        "--privileged",
+        // The container needs to access the host's PID namespace to mount host directories
+        "--pid=host",
+        // Since https://github.com/containers/bootc/pull/919 this mount should not be needed, but
+        // some reason with e.g. quay.io/fedora/fedora-bootc:41 it is still needed.
+        "-v",
+        "/var/lib/containers:/var/lib/containers",
+        // TODO: Get from argv
+        "quay.io/fedora/fedora-bootc:41",
+        // We're replacing the current root
+        "bootc",
+        "install",
+        "to-existing-root",
+        // The user already knows they're reinstalling their machine, that's the entire purpose of
+        // this binary. Since this is no longer an "arcane" bootc command, we can safely avoid this
+        // timed warning prompt. TODO: Discuss in https://github.com/containers/bootc/discussions/1060
+        "--acknowledge-destructive",
+    ];
+
+    Command::new(command_and_args[0])
+        .args(&command_and_args[1..])
+        .status()
```

Comment:
At this point we could consider just `exec` too?

Reply:
I kinda want to print the command at the end, in case it fails. But otherwise yeah exec makes perfect sense

Reply:
We can still print an error if the exec fails right? There's several other instances of `exec()` in this project.

Reply:
>We can still print an error if the exec fails right? 

Isn't the point of `exec` that the `podman` process replaces the `bootc-installer` process? Nothing can run after an `exec`. Am I misunderstanding?

Reply:
As far as I can tell right now, nothing is running after the `Command`? Hence we could use exec. But it's a really minor point honestly.

Reply:
The `?` operator is running after the command. Thus:

> kinda want to print the command at the end, in case it fails.

Reply:
Yes, handling the error from exec requires some special handling because it always returns an error (it never returns on success). Here's an example of that in this project:
https://github.com/containers/bootc/blob/5f0dd2361a8ba33ff339359676eb006f3cc8c28d/lib/src/install.rs#L881

Reply:
Oh I didn't realize it returns on error. I thought it never returns period. Will do that

Reply:
Oh and it only errors if there was a problem running the command itself. If the command that was ran fails (non-zero exit code), it will not return an error

Reply:
I'd say we should probably print the command unconditionally before we start, regardless of error or not; WDYT?

Reply:
Definitely, but by that point it will be buried under 50 layer podman pull logs. So I thought it would be nice to have it after